File: baseproject-android/app/src/main/java/com/chang/template/MyApplication.java
package com.chang.template;

import android.content.Context;
import android.net.Uri;
import android.support.multidex.MultiDexApplication;


import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * Created by Howard Chang on 2016/11/29
 */
public class MyApplication extends MultiDexApplication {


    @Override
    public void onCreate() {
        super.onCreate();

        buildPicasso(this);
    }

    public static void buildPicasso(Context context) {

    }

}

File: baseproject-android/app/src/main/java/com/chang/template/MyApplication.kt
package com.chang.template

import androidx.multidex.MultiDexApplication

import com.chang.template.AppModule.mModule
import com.crashlytics.android.Crashlytics
import com.facebook.stetho.Stetho
import io.fabric.sdk.android.Fabric
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin

/**
 * Created by Howard Chang on 2016/11/29
 */
class MyApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.FLAVOR.toLowerCase() != "publish") {
            Stetho.initializeWithDefaults(this)
        } else {
            Fabric.with(this, Crashlytics());
        }

        startKoin {
            // Android context
            androidContext(this@MyApplication)
            // modules
            modules(mModule)
        }
    }

}
